apps/src/aiTutor/views/AssistantMessage.tsx

```diff
@@ -0,0 +1,87 @@
+import React, {useState} from 'react';
+import classNames from 'classnames';
+
+import {ChatCompletionMessage} from '@cdo/apps/aiTutor/types';
+import Typography from '@cdo/apps/componentLibrary/typography/Typography';
+import Button from '@cdo/apps/templates/Button';
+
+import {saveFeedback, FeedbackData} from '../interactionsApi';
+import style from './chat-workspace.module.scss';
+
+interface AssistantMessageProps {
+  message: ChatCompletionMessage;
+}
+
+const AssistantMessage: React.FC<AssistantMessageProps> = ({message}) => {
+  const [feedbackState, setFeedbackState] = useState<FeedbackData>({
+    thumbsUp: false,
+    thumbsDown: false,
+  });
+
+  const handleFeedbackSubmission = async (
+    thumbsUp: boolean,
+    messageId?: number
+  ) => {
+    if (!messageId) {
+      return;
+    }
+
+    const feedbackData = {
+      thumbsUp: thumbsUp ? (feedbackState.thumbsUp ? null : true) : null,
```

Comment:
why is it if `feedbackState.thumbsUp` is true we set `thumbsUp` to `null`?

Reply:
If we're clicking thumbs up again (`thumbsUp` === `true`) and the existing state (`feedbackState.thumbsUp`) is true, then we want to nullify the provided feedback value to "toggle" it to its original state.

Reply:
The truth table should be:

| thumbsUp (clicked thumbs up) | feedbackState.thumbsUp (already clicked?) | Result (new selected value)                         |
|----------|------------------------|---------------------------------|
| F        | F                      | null                            |
| F        | T                      | null                            |
| T        | F                      | true                            |
| T        | T                      | null                            |

Reply:
Ah I see thanks! Can you add a comment that this is a toggle value?